### ISSUE TITLE
Add inline text translation

### ISSUE THREAD

**Issue by patrickgold** [florisboard/florisboard#1871]
This issue sums up the proposals for a possible addition of text translation directly into the keyboard to avoid duplicate discussions. *All other issues requesting the addition of a translator will be closed as a duplicate. Please use this issue for discussion around adding a translator.*

### State of this proposal

Currently it is undecided if a translator will be included directly into this keyboard, as this is a massive effort and it is not easily finding an existing implementation which directly fits into this project. However I am always open to add one in the future (>0.5.0) if there's a good way of implementing such translator while fully respecting the privacy of the user.

### Requirements for possible translation projects/services

Preferably, the translation works fully offline and the translation data never leaves the device. Updating the translation dictionaries may be done via network requests, if needed. The translation library must not depend on the GMS services. For this project to be usable, it should either be written in native C/C++ code or in Java/Kotlin.

If you know (or have heard of) a good open-source offline-translation library, please comment a link to the repository or project web page as a comment below, this would really help!

---

Reference issues:
#933 
#1456 
#1865

**Comment by Linerly** [comment]
I'm thinking that [LibreTranslate](https://github.com/LibreTranslate/LibreTranslate), or the [Argos Translate](https://github.com/argosopentech/argos-translate) Python library (which provides the translations on LibreTranslate), or [Lingva Translate](https://github.com/TheDavidDelta/lingva-translate) (a front-end to Google Translate) could be used in FlorisBoard

**Comment by my-garbage** [comment]
Here is an easy way that solves the translation problem offline, privacy respecting and also memory saving as it also allows replacement of Google lens and traditional document translation with the same models.

https://github.com/DavidVentura/offline-translator/issues/170

Use the app/src/main/java/dev/davidv/translator/AidlTranslationService.kt for your translation needs the keyboard just needs to have a Textfeld and 2 dropdown selects for languages.

**Comment by my-garbage** [comment]
Using it that way also gives the opportunity to be able to support all the other translation tools too meanwhile the heavy lifting of good translation is away from the making of florisboard.

**Comment by hasanelfalakiy** [comment]
Is there any update on this feature?

**Comment by Nekothegamer** [comment]
yeah, any update on this? if this gets releasd into florisboard I'll switch in a heartbeat, but until then i have to stay with Gboard for this feature...

**Comment by hasanelfalakiy** [comment]
> yeah, any update on this? if this gets releasd into florisboard I'll switch in a heartbeat, but until then i have to stay with Gboard for this feature...

Is your Gboard experiencing the same issue where recently copied text cannot be pasted from the suggestion bar? Like this:
https://github.com/user-attachments/assets/493fde3c-5bed-48fc-b2cd-0275bb56552b

But there are no issues with FlorisBoard:
https://github.com/user-attachments/assets/a94f7c30-f444-44ae-90c1-3045954d2686
